### PR TITLE
docs: subdaemon debugging

### PR DESCRIPTION
**Pull request by jb55**

### CODE REVIEW

File: common/subdaemon.c [hunk 1]
```diff
@@ -83,7 +83,7 @@ void subdaemon_setup(int argc, char *argv[])
 	for (i = 1; i < argc; i++) {
 		while (streq(argv[i], "--debugger")) {
 			if (!printed)
-				fprintf(stderr, "gdb -ex 'attach %u' %s\n",
+				fprintf(stderr, "gdb -ex 'attach %u' %s -ex return -ex cont\n",
```
Comment (cdecker) [inline common/subdaemon.c]: Returning once is not always sufficient, it may be in the `streq` call when attaching.

**Comment by jb55** [comment]
> Returning once is not always sufficient, it may be in the `streq` call when attaching.

Ah I've never run into this case yet. I've updated the docs to reflect
this. I assume we can't safely do `-ex return -ex return -ex cont`, so I
only print out one `-ex return` and the document the remaining commands
in HACKING.

**Comment by cdecker** [comment]
If we make it an empty loop we should be at the correct level, isn't it?

**Comment by jb55** [comment]
> If we make it an empty loop we should be at the correct level, isn't it?

Ah good call

**Comment by jb55** [comment]
I can't seem to get it working. The compiler does something funky with
inlining the fprintf call, and whenever I attach it seems to always be
inside of it, saying you can't return from an inlined function call!?

I'm going to try some other things.

**Comment by jb55** [comment]
Couldn't get anything else working, I'm happy with this patch for now.

**Comment by rustyrussell** [comment]
We should spin on a volatile variable.  Then you set it to true to break out?

**Comment by rustyrussell** [comment]
Is that better?  @jb55 please review...

**Comment by cdecker** [comment]
Nice one @rustyrussell, I assume the debugger still waits for a `continue`, but will break out of the loop when we do?

ACK 694035a

**Comment by jb55** [comment]
Works much better. Thanks!

Tested ACK 694035a547ac2a32147f3739aff33295303c0761

**Comment by rustyrussell** [comment]
Yes, I didn't put 'continue' in the cmdline, since you may want to set breakpoints, etc.

**Pull request merged**